cli/internal/fs/turbo_json.go

```diff
@@ -248,82 +293,169 @@ func (pc Pipeline) HasTask(task string) bool {
 	return false
 }
 
-// UnmarshalJSON deserializes JSON into a TaskDefinition
-func (c *TaskDefinition) UnmarshalJSON(data []byte) error {
+// Pristine returns a PristinePipeline
+func (pc Pipeline) Pristine() PristinePipeline {
+	pristine := PristinePipeline{}
+	for taskName, taskDef := range pc {
+		pristine[taskName] = taskDef.TaskDefinition
+	}
+	return pristine
+}
+
+// hasField checks the internal bookkeeping fieldsMeta field to
+// see whether a field was actually in the underlying turbo.json
+// or whether it was initialized with its 0-value.
+func (btd BookkeepingTaskDefinition) hasField(fieldName string) bool {
+	if _, ok := btd.fieldsMeta[fieldName]; ok {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	_, ok := btd.fieldsMeta[fieldName]
        return ok
```

Reply:
I prefer explicit true/false. I don't want to guess the value of `ok` in golang (even though i can guess it based on the return type of the function)